### PR TITLE
Feature/fwf 3620 history component

### PR DESCRIPTION
**Pull request by Ajay-aot**
# Issue Tracking

JIRA: https://aottech.atlassian.net/browse/FWF-3620
https://aottech.atlassian.net/browse/FWF-3644
https://aottech.atlassian.net/browse/FWF-3726
https://aottech.atlassian.net/browse/FWF-3762
Issue Type: BUG/ FEATURE

# Changes
reusable history modal component implemented

#Screenshots
image
image

### CODE REVIEW

File: forms-flow-theme/scss/historyModal.scss [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      padding: 0.688rem var(--spacer-100);
+      border-radius: var(--radius-0150);
+      border: 2px solid var(--ff-gray-100);
+      @extend .version-grid;
+      margin-left: 2rem;
+      grid-template-columns: 0.77fr 1fr 1fr 1fr 1fr 1fr;
+    }
+
+    .bold-headings {
+      color: var(--ff-gray-800, #303436) !important;
```
Comment (arun-s-aot) [inline forms-flow-theme/scss/historyModal.scss:116]: remove default variables in full file

File: forms-flow-theme/scss/historyModal.scss [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    .workflow-minor-grid {
+      padding: 0.688rem var(--spacer-100);
+      border-radius: var(--radius-0150);
+      border: 2px solid var(--ff-gray-100);
+      @extend .version-grid;
+      margin-left: 2rem;
+      grid-template-columns: 0.77fr 1fr 1fr 1fr 1fr 1fr;
+    }
+
+    .bold-headings {
```
Comment (arun-s-aot) [inline forms-flow-theme/scss/historyModal.scss:115]: can u rename this

File: forms-flow-theme/scss/historyModal.scss (hunk 1 above)
Comment (Ajay-aot) [reply forms-flow-theme/scss/historyModal.scss]: done

File: forms-flow-theme/scss/historyModal.scss (hunk 2 above)
Comment (Ajay-aot) [reply forms-flow-theme/scss/historyModal.scss]: done

File: forms-flow-components/src/components/CustomComponents/HistoryModal.tsx [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+                      variant="secondary"
+                      size="sm"
+                      label={loadMoreBtnText}
+                      onClick={handleLoadMore}
+                      dataTestid={loadMoreBtndataTestid}
+                      ariaLabel={loadMoreBtnariaLabel}
+                    />
+                  </div>
+                )}
+            </div>
```
Comment (arun-s-aot) [inline forms-flow-components/src/components/CustomComponents/HistoryModal.tsx:326]: seems like these are duplicate code. Can u pls check @Ajay-aot

File: forms-flow-components/src/components/CustomComponents/HistoryModal.tsx [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+        const isMajorVersion = entry.isMajor;
+        const version = `${entry.majorVersion}.${entry.minorVersion}`;
+        const cloned_form_id =
+          categoryType === "FORM" ? entry.changeLog.cloned_form_id : null;
+        const process_id = categoryType === "WORKFLOW" ? entry.id : null;
+        const isLastEntry = index === allHistory.length - 1;
+        return (
+          <React.Fragment key={`${entry.version}-${index}`}>
+            {isMajorVersion && (
+              <div
```
Comment (arun-s-aot) [inline forms-flow-components/src/components/CustomComponents/HistoryModal.tsx:201]: From line 200 to 263, can we reduce the duplication . 
Use any AI tools if needed to handle this kinds of optimisations. but make sure you review it and add @Ajay-aot

**Review by arun-s-aot — CHANGES_REQUESTED** [review]
Please check the review comments and update the code

File: forms-flow-components/src/components/CustomComponents/HistoryModal.tsx (hunk 3 above)
Comment (Ajay-aot) [reply forms-flow-components/src/components/CustomComponents/HistoryModal.tsx]: updated

File: forms-flow-components/src/components/CustomComponents/HistoryModal.tsx (hunk 4 above)
Comment (Ajay-aot) [reply forms-flow-components/src/components/CustomComponents/HistoryModal.tsx:234]: updated

File: forms-flow-components/src/components/CustomComponents/HistoryModal.tsx [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+  const formattedHours = String(displayHours).padStart(2, "0");
+
+  return `${day}-${month}-${year} ${formattedHours}:${minutes}${ampm}`;
+};
+
+const HistoryField = ({ heading, value }) => {
+  return (
+    <div>
+      <div className="content-headings">{heading}</div>
+      <div className="normal-text">{value}</div>
```
[the remark this answers is not in the document]
Comment (Ajay-aot) [reply forms-flow-components/src/components/CustomComponents/HistoryModal.tsx]: updated

**Review by arun-s-aot — APPROVED** [review]

**Review by fahad-aot — APPROVED** [review]

**Review by Josephalexantony-aot — APPROVED** [review]

**Pull request merged**